src/encode.py

```diff
@@ -3,61 +3,73 @@
 
 from tqdm import tqdm
 
-FEATURES_DEFAULT = ('lesson_id', 'exercise_code', 'learning_object', 'exercise_code_level')
 TIME_WINDOWS_DEFAULT = ('60s', '1h', '1d', '5d', '30d', '365d')
 COUNTERS_DEFAULT = ('attempts', 'wins')
 
-def encode_df(df: pd.DataFrame, features: tuple=FEATURES_DEFAULT, counters: tuple=COUNTERS_DEFAULT, time_windows: tuple=TIME_WINDOWS_DEFAULT)-> pd.DataFrame:
+
+def encode_df(df: pd.DataFrame, exercises: tuple = None, counters: tuple = COUNTERS_DEFAULT, time_windows: tuple = TIME_WINDOWS_DEFAULT) -> pd.DataFrame:
```

Comment:
for readability i think you should limit your line length, for instance where the indication line is in pycharm
even if it's more lines, it might be easier to read like this:
```
def encode_df(
    df: pd.DataFrame, 
    exercises: tuple = None, 
    counters: tuple = COUNTERS_DEFAULT, 
    time_windows: tuple = TIME_WINDOWS_DEFAULT
) -> pd.DataFrame:
```

Reply:
also i think you should initialize the `exercises` variable with an empty tuple `()`
or put Optional[Tuple] as a type and let it = None.

Reply:
done